leaflet/index.d.ts

```diff
@@ -817,67 +817,72 @@ declare namespace L {
         coordsToLatLng?: (coords: [number, number] | [number, number, number]) => LatLng; // check if LatLng has an altitude property
     }
 
-    /**
-     * Represents a GeoJSON object or an array of GeoJSON objects.
-     * Allows you to parse GeoJSON data and display it on the map. Extends FeatureGroup.
-     */
-    export interface GeoJSON extends FeatureGroup {
-        /**
-         * Adds a GeoJSON object to the layer.
-         */
-        addData(data: GeoJSON.GeoJsonObject): Layer;
-
-        /**
-         * Resets the given vector layer's style to the original GeoJSON style,
-         * useful for resetting style after hover events.
-         */
-        resetStyle(layer: Layer): Layer;
-
-        /**
-         * Changes styles of GeoJSON vector layers with the given style function.
-         */
-        setStyle(style: StyleFunction): this;
-
+    export class GeoJSON {
         /**
          * Creates a Layer from a given GeoJSON feature. Can use a custom pointToLayer
          * and/or coordsToLatLng functions if provided as options.
          */
-        geometryToLayer(featureData: GeoJSON.Feature<GeoJSON.GeometryObject>, options?: GeoJSONOptions): Layer;
+        static geometryToLayer(featureData: GeoJSON.Feature<GeoJSON.GeometryObject>, options?: GeoJSONOptions): Layer;
 
         /**
          * Creates a LatLng object from an array of 2 numbers (longitude, latitude) or
          * 3 numbers (longitude, latitude, altitude) used in GeoJSON for points.
          */
-        coordsToLatLng(coords: [number, number]): LatLng;
-
-        coordsToLatLng(coords: [number, number, number]): LatLng;
+        static coordsToLatLng(coords: [number, number] | [number, number, number]): LatLng;
 
         /**
          * Creates a multidimensional array of LatLngs from a GeoJSON coordinates array.
          * levelsDeep specifies the nesting level (0 is for an array of points, 1 for an array of
          * arrays of points, etc., 0 by default).
          * Can use a custom coordsToLatLng function.
          */
-        coordsToLatLngs(coords: Array<number>, levelsDeep?: number, coordsToLatLng?: (coords: [number, number] | [number, number, number]) => LatLng): LatLng[]; // Not entirely sure how to define arbitrarily nested arrays
+        static coordsToLatLngs(
+            coords: any[],
+            levelsDeep?: number,
+            coordsToLatLng?: (coords: [number, number] | [number, number, number]) => LatLng): any[]; // Using any[] to avoid artificially limiting valid calls
 
         /**
          * Reverse of coordsToLatLng
          */
-        latLngToCoords(latlng: LatLng): [number, number] | [number, number, number];
+        static latLngToCoords(latlng: LatLng): number[];
```

Comment:
With the old definition, the following code gives a compiler error:

```js
let threeCoords: [number, number, number] = [1, 2, 3];
latLng = L.GeoJSON.coordsToLatLng(threeCoords);
threeCoords = L.GeoJSON.latLngToCoords(latLng);
```

With this error:
error TS2322: Type '[number, number] | [number, number, number]' is not assignable to type '[number, number, number]'.
  Type '[number, number]' is not assignable to type '[number, number, number]'.
    Property '2' is missing in type '[number, number]'.

The typescript docs (https://www.typescriptlang.org/docs/handbook/advanced-types.html) state that "If a value has the type A | B, we only know for certain that it has members that both A and B have." which means that we can only assign the return value of the 'coordsToLatLng' function to [number, number] because the the [number, number, number] tuple has a property '2' (the third value in the tuple) that doesn't exist on the [number, number] tuple. The union type works ok in the parameter definition because the constraint is on the consumer of the type, not the provider of the type.

Does that make sense? I'm not an expert in Typescript, but I ran into the issue when I added the GeoJSON stuff to the tests and then found that documentation on the Typescript docs site.